sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py

```diff
@@ -0,0 +1,309 @@
+
+    return expected_df
+
+
+@parametrize_offline_retrieval_test
+def test_historical_features(environment: Environment):
+    store = environment.feature_store
+
+    customer_df, customer_fv = (
+        environment.customer_df,
+        environment.customer_feature_view(),
+    )
+    driver_df, driver_fv = (
+        environment.driver_df,
+        environment.driver_stats_feature_view(),
+    )
+    orders_df = environment.orders_df
+    full_feature_names = environment.test_repo_config.full_feature_names
+
+    entity_df_query = None
+    if environment.orders_table():
+        entity_df_query = f"SELECT * FROM {environment.orders_table()}"
+
+    event_timestamp = (
+        DEFAULT_ENTITY_DF_EVENT_TIMESTAMP_COL
+        if DEFAULT_ENTITY_DF_EVENT_TIMESTAMP_COL in orders_df.columns
+        else "e_ts"
+    )
+    expected_df = get_expected_training_df(
+        customer_df,
+        customer_fv,
+        driver_df,
+        driver_fv,
+        orders_df,
+        event_timestamp,
+        full_feature_names,
+    )
+
+    if entity_df_query:
+        job_from_sql = store.get_historical_features(
+            entity_df=entity_df_query,
+            features=[
+                "driver_stats:conv_rate",
+                "driver_stats:avg_daily_trips",
+                "customer_profile:current_balance",
+                "customer_profile:avg_passenger_count",
+                "customer_profile:lifetime_trip_count",
+            ],
+            full_feature_names=full_feature_names,
+        )
+
+        start_time = datetime.utcnow()
+        actual_df_from_sql_entities = job_from_sql.to_df()
+        end_time = datetime.utcnow()
+        print(
+            str(f"\nTime to execute job_from_sql.to_df() = '{(end_time - start_time)}'")
+        )
+
+        assert sorted(expected_df.columns) == sorted(
+            actual_df_from_sql_entities.columns
+        )
+
+        actual_df_from_sql_entities = (
+            actual_df_from_sql_entities[expected_df.columns]
+            .sort_values(by=[event_timestamp, "order_id", "driver_id", "customer_id"])
+            .drop_duplicates()
+            .reset_index(drop=True)
+        )
+        expected_df = (
+            expected_df.sort_values(
+                by=[event_timestamp, "order_id", "driver_id", "customer_id"]
+            )
+            .drop_duplicates()
+            .reset_index(drop=True)
+        )
+
+        assert_frame_equal(
+            actual_df_from_sql_entities, expected_df, check_dtype=False,
+        )
+
+        table_from_sql_entities = job_from_sql.to_arrow()
+        df_from_sql_entities = (
+            table_from_sql_entities.to_pandas()[expected_df.columns]
+            .sort_values(by=[event_timestamp, "order_id", "driver_id", "customer_id"])
+            .drop_duplicates()
+            .reset_index(drop=True)
+        )
+        assert_frame_equal(actual_df_from_sql_entities, df_from_sql_entities)
+
+    # timestamp_column = (
```

Comment:
remove these?

Reply:
or if you want to fix it, consider moving into separate test as I suggest below

Reply:
I'm going to remove them because the behaviour isn't actually consistent across offline stores. File sources don't support sql queries at all.